### DISCUSSION TITLE
Passing spanId when starting to create a Span with SDKSpanBuilder

### DISCUSSION THREAD

**Discussion by honho001** [open-telemetry/opentelemetry-java#5995] [Q&A] [answered]
In the **SDKSpanBuilder** class, when the **startSpan** function is called. It typically generates the **spanId** internally.
Is there any way to pass the **spanID** when starting to create a Span?

**Comment by jkwatson** [comment]
There is not. What is your use-case?

**Reply by honho001** [reply]
I have some servers that will send logs to my Java app.  In these logs, there is some information such as globalId, callerId, and localId.
 In my Java app i use Otel to collect these logs.
 I plan to convert this information as follows: 
 + globalId to traceId
 + localId to spanId
 + callerId to parentSpanId
 
But when I started doing that, I noticed that Otel does not allow me to customize the spanId.
As a newbie to Otel, I'm dn't know if there is a way to achieve my idea.

**Reply by jkwatson** [reply, marked as answer]
OTel has very strict requirements for the structure of a span ids, so this isn't something that is supported. Better would probably be to make these attributes of your spans, rather than trying to make them into the otel ids.

**Reply by honho001** [reply]
thank you
